components/core/src/util/posix_perm.rs

```diff
@@ -60,6 +60,25 @@ pub fn set_owner<T: AsRef<Path>, X: AsRef<str>>(path: T, owner: X, group: X) ->
     }
 }
 
+pub fn ensure_path_permissions(path: &Path, permissions: u32) -> Result<()> {
```

Comment:
Great call on the pub(crate). I agree on the side effect part of your comment but I think I disagree on the overkill part.  If the permissions at any point in the directory hierarchy prevent read or execute then we won't have solved the issue in the eyes of the customer.  I kind of feel like there are some many potential unknowns that a more active hand is called for.  Substantially taking your suggested comment I've gone with the following for now.

```
// This is required on machines where umask is set to a higher value like `0077`. (See CHEF-10987)
// This has a side effect of potentially changing the *mode* of directories not created by us but this
// is done in order to ensure the ability to execute in the face of the variety of scenarios that may
// be encounter "in the wild". This is as such not a huge problem as we are only changing the *mode*
// for `ancestors` we are owner of.
```

Reply:
In our daily today we agreed that a comment discussing was the most important thing. I think we're good but I'd prefer you to eyeball it one last time and mark resolved instead of me steamrolling it in.

Reply:
When I said, avoiding that `ancestors` call is possible but just too much of an effort which is an overkill. Here is how it may look like - 

1. Take the parts of the path
2. for each of the part recursively call - `create_dir` and `chown` 

This will avoid having to `chown` on say `/hab/pkgs` if it is owned by the `user` and has some other permissions (which are unlikely in most scenarios). 

The way it is currently done is just fine - once we add the comment.